tests/publiccloud/ipa.pm

```diff
@@ -16,6 +16,20 @@ use strict;
 use warnings;
 use testapi;
 
+sub extract_time {
+    my ($string, $regex) = @_;
+    my $extract_sec     = qr/(\d{1,2}\.\d{1,3})/;
+    my $extract_minutes = qr/(\d{1,2})min (\d{1,2}\.\d{1,3})/;
+    my @rez             = $string =~ /$regex/;
+    if (defined $rez[1]) {
```

Comment:
I wonder if we can avoid rely on a regex which comes as parameter.

What about a function like:
```
sub parse_system_analyze {
    my $line = shift;
    my $res = {};
    $line =~ s/Startup finished in\s*//;
    $line =~ s/=(.+)$/+$1 (overall)/;
    for my $time (split(/\s*\+\s*/, $line)){
        if ($time =~ /((\d{1,2})\s*min\s*)?(\d{1,2}\.\d{1,3})s\s*\((\w+)\)/ ){
            my $sec = $3;
            $sec += $2 * 60 if (defined($1));
            $res->{$4} = $sec;
        }
    }
    map {die($_) unless exists($res->{$_})} qw(kernel initrd userspace overall);
    return $res;
}
```

gives you the results as:
```
{
          'userspace' => '39.733',
          'initrd' => '23.946',
          'overall' => '81.09',
          'kernel' => '17.409'
        };
```
and we can easily make statistics on all different measurements.

Reply:
1. in this I would avoid creating a function and just add this code inline 
2. "I wonder if we can avoid rely on a regex which comes as parameter." - it sounds that you think that passing regex as parameter is bad thing . I wonder why ?
3. can't understand all things in your regex: 
 -`s/` - google saying that it forces regex to treat multiline as single line but it **is** single line we know it because we rely on systemd-analyze output format 
  - why you decide to expect spaces between (`1min`) when we **know** that there are none .
But probably before diving into such details I want to clarify one point - do we really need userspace and initrd timings ? First of all this PR suppose to just fix issue with cases when one kernel or overall timings takes longer than minute. Second in my PR I compare kernel and overall timings with some dummy limits , do you want me to do the same for userspace and initrd ?

Reply:
For me its just a different representation of data and I wonna have it parsed to further work with that data. So for me having one function which get a string and return a struct is the best way.
I don't wonna check dummy limits for initrd and userspace. But if we are going to store such values in a graphana db, I would do so.

1) Why it's more readable to give this code section a name and I think it's easy to understand having a function, gets an input and produce a defined output.

2) Rely on the groups number isn't that good. If you need to change the regex or make some bigger part with a quantifier, the group number change quite easy. So if you need to rely on such things, I would name these groups.

3) `s///` is the substitution operator.